content/blog/2024/10/kubewarden-1-17-release.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Kubewarden 1.17 release
+authors:
+  - Víctor Cuadrado Juan
+date: 2024-09-27
+---
+
+We are thrilled to announce the release of Kubewarden v1.17.0. This release is
+packed with big features, let's have a look!
+
+## Certificate rotation & removal of cert-manager dependency
+
+Starting from this release, the Kubewarden stack takes care of creating and
+rotating all the needed TLS certificates and certificate authorities.
+
+Kubewarden, by virtue of connecting to the Kubernetes API server, needs TLS
+certificates for both the kubewarden-controller (when creating webhooks for its
+policies) and for the PolicyServers (so they can report their results to the
+Webhook API server).
+
+Up until v1.16, the certificate authority (CA) and leaf certificates used by
+kubewarden-controller to connect to the Webhooks API were generated by
+cert-manager.
+The certificates used to register out PolicyServers with the Kubernetes Webhook
+API were and continue to be generated by our controller, and our controller
+generates a self-signed Certificate Authority that is used to sign all the
+PolicyServer certs. These had a 10 year expiration date. Certificate rotation
+for all was left to the user.
+
+From v1.17 onwards, we generate the CA and certificate on Helm install or
+upgrade if not present, under `kubewarden-ca` Secret. This means we don't
+depend on cert-manager anymore. The CA defaults to a 10 year expiration date,
+and the leaf certificates to a 1 year expiration date.
+
+In addition, now the kubewarden-controller rotates both the CA and the needed
+certificates for the webhooks and PolicyServers on its own, 1 month before the
+expiration date. Care is taken to ensure a valid overlap of old and new
+certificates, and removal of old certificates as soon as possible for security
+reasons.
+
+Hence, cert-manager can now be removed if it was installed just because of
+Kubewarden. In case it is being used for other workloads, it can be left
+installed.
```

Comment:
```suggestion
So, cert-manager can now be removed if it was installed as a dependency of
Kubewarden. It can be left installed if it is being used for other workloads.
```